build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -230,6 +232,12 @@ class WatchImpl implements BuildState {
   /// Checks if we should skip a watch event for this [change].
   bool _shouldProcess(AssetChange change) {
     assert(_assetGraph != null);
+    if (_isRootPackagesFile(change.id)) {
+      _fatalBuildCompleter.complete();
```

Comment:
Is there a way we can structure this without adding side effects to this method?

Reply:
I am looking into it, but without setting up an additional watcher it is not clear what the best way would be.

Reply:
I moved it up earlier so at least it isn't so hidden inside of `_shouldProcess`, lmk what you think.

Reply:
I like the new location inside the `asyncMap` call.